x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.ts

```diff
@@ -116,6 +117,7 @@ export const baseSchema = buildRuleSchemas({
     output_index: AlertsIndex,
     namespace: AlertsIndexNamespace,
     meta: RuleMetadata,
+    custom_highlighted_fields: RuleCustomHighlightedFieldArray,
```

Comment:
Posting for visibility here. As discussed over zoom with @yctercero, I think this field should be:

- Defaultable instead of optional, meaning it should always be defined in `RuleResponse` and other types of rules we return from the API (`QueryRule`, `ThresholdRule`, etc).
- Wrapped in an object to make space for other parameters we might want to add for this feature in the future. This would also make it more compatible with the `DiffableRule` interface and the rule diff/upgrade algorithm.
- Renamed to something more UI-neutral and generic, e.g. "investigation fields".

So in a rule, it would be something like `rule.investigation_fields.field_names`.

We should consider writing a migration to add a "null" `investigation_fields` object to existing rules. Pseudocode:

```ts
rule.investigation_fields = {
  field_names: [],
};
```